### DISCUSSION TITLE
How should we sort links in internal playlists?

### DISCUSSION THREAD

**Discussion by freearhey** [iptv-org/community#527] [Polls]
In this case we are talking only about internal playlists located in the [/streams](https://github.com/iptv-org/iptv/tree/master/streams) folder.

By __tvg-id__:

```m3u
#EXTINF:-1 tvg-id="",SBS News
https://ythls.onrender.com/channel/UCkinYTS9IHqOEwR1Sze2JTw.m3u8
#EXTINF:-1 tvg-id="HLANDTV.kr",MBC Chuncheon
https://stream.chmbc.co.kr/TV/myStream/playlist.m3u8
#EXTINF:-1 tvg-id="HLKUDTV.kr",MBC Busan
https://stream.bsmbc.com/livetv/BusanMBC_TV_onairstream/playlist.m3u8
```

By __channel name__:

```m3u
#EXTINF:-1 tvg-id="HLKUDTV.kr",MBC Busan
https://stream.bsmbc.com/livetv/BusanMBC_TV_onairstream/playlist.m3u8
#EXTINF:-1 tvg-id="HLANDTV.kr",MBC Chuncheon
https://stream.chmbc.co.kr/TV/myStream/playlist.m3u8
#EXTINF:-1 tvg-id="",SBS News
https://ythls.onrender.com/channel/UCkinYTS9IHqOEwR1Sze2JTw.m3u8
```

After the survey is over, the selected rule will be added to the [Contribution Guide](https://github.com/iptv-org/iptv/blob/master/CONTRIBUTING.md) and the script responsible for sorting will be updated accordingly.

PR: https://github.com/iptv-org/iptv/pull/14226

**Comment by BellezaEmporium** [comment]
Decided to go to what seemed logical to me, aka "Channel name". The fact that there could be "call signs" could disorganise the overview of the channels list (the example of the KXXX and WXXX in the US is a great one).

**Reply by brokenstream** [reply]
I voted "Channel name" for the same reason. Sorting by ID causes channels from networks like NBC and FOX, as well as many "Canal" names, to spread out among their respective playlists due to the various callsigns, making them more tedious to search for.

**Comment by Alstruit** [comment]
I prefer tvg-id since it might be easier to search and replace the links.
